### PR TITLE
switch to devicemapper-rs 4.5

### PR DESCRIPTION
**Pull request by trgill**
Update Cargo.toml and Cargo.lock
Handle errors from devicemapper-rs 4.5 (previously Stratis errors)
Adjust to LinearDev now taking Segments as a parameter
Move filesytems related functions to filesystem.rs
Update use statements to include moved structs
Remove no longer used/needed files
Device::path() changed to Device::devnode() - made adjustments
Update integration tests

### CODE REVIEW

**Comment by trgill** [comment]
@agrover please review

File: src/dbus_api/util.rs [hunk 1]
```diff
@@ -29,7 +29,8 @@ pub fn get_next_arg<'a, T>(iter: &mut Iter<'a>, loc: u16) -> Result<T, MethodErr
     if iter.arg_type() == ArgType::Invalid {
         return Err(MethodErr::no_arg());
     };
-    let value: T = try!(iter.read::<T>().map_err(|_| MethodErr::invalid_arg(&loc)));
+    let value: T = try!(iter.read::<T>()
```
Comment (agrover) [inline src/dbus_api/util.rs]: Looks like some whitespace changes were accidentally included here in 4a178928.

File: src/types.rs [hunk 2]
```diff
@@ -39,6 +40,7 @@ pub enum StratisError {
     Nix(nix::Error),
     Dbus(dbus::Error),
     Term(term::Error),
+    DeviceMapper(devicemapper::result::DmError),
```
Comment (agrover) [inline src/types.rs]: How about DM instead of DeviceMapper?

File: src/engine/strat_engine/blockdev.rs [hunk 3]
```diff
@@ -14,10 +14,10 @@ use std::str::FromStr;
 use std::thread;
 use std::time::Duration;
 
+use devicemapper::consts::SECTOR_SIZE;
 
 use devicemapper::Device;
```
Comment (agrover) [inline src/engine/strat_engine/blockdev.rs]: OK, but why the space between consts and Device? They can go together I think.

Comment (agrover) [inline src/engine/strat_engine/blockdev.rs]: I think this was taken care of by the last commit, so disregard.

File: src/types.rs (hunk 2 above)
Comment (trgill) [inline src/types.rs]: Ok

File: src/dbus_api/util.rs (hunk 1 above)
Comment (trgill) [inline src/dbus_api/util.rs]: Fixed.

File: src/engine/strat_engine/blockdev.rs (hunk 3 above)
Comment (trgill) [inline src/engine/strat_engine/blockdev.rs]: Ok

**Pull request merged**